### ISSUE TITLE
Failed to deserialize string to enum

### ISSUE THREAD

**Issue by ongchi** [twking7/fixed_width#4]
I'm trying to deserialize string to enum, like this:

```rust
#[derive(FixedWidth, Deserialize, Debug)]
struct Foo {
    #[fixed_width(range = "0..3")]
    bar: Bar,
}

#[derive(Deserialize, Debug)]
enum Bar {
    Foo,
    Bar,
    Baz,
}

let foo: Foo = fixed_width::from_str("Baz").unwrap();
println!("{:?}", foo);
```

expected to get the struct:

```
Foo { bar: Baz }
```

but get error message:

```
DeserializeError(Message("unknown variant `bar`, expected one of `Foo`, `Bar`, `Baz`"))
```

**Comment by ongchi** [comment]
I found this works when derive Deserialize from FromStr, as following:

```rust
use serde_with::DeserializeFromStr;

#[derive(DeserializeFromStr, Debug)]
enum Bar {
    Foo,
    Bar,
    Baz,
}

impl FromStr for Bar {
    type Err = String;

    fn from_str(s: &str) -> Result<Self, String> {
        match s {
            "Foo" => Ok(Self::Foo),
            "Bar" => Ok(Self::Bar),
            "Baz" => Ok(Self::Baz),
            _ => Err("errr!".to_string()),
        }
    }
}
```

It's works as expected but too much boilerplate code.
Would you have a suggestion?

**Comment by twking7** [comment]
Hello @ongchi , I believe this happens because by default, serde expects the serialized value to be capitalized, i.e. it expects `Bar`.

If you can control the serialized value, that may be one way to fix this. If you cannot, try adding this to your enum type:

`#[serde(rename_all = "lowercase")]`

**Comment by ongchi** [comment]
Hi @twking7 , thank you for your reply.

I got deserialized struct after adding `#[serde(rename_all = "lowercase")]` to enum type.
But it always deserialize to `Bar::Bar`. That's strange!

```rust
let foo: Foo = fixed_width::from_str("Bar").unwrap();
```
give us `Foo { bar: Bar }`

```rust
let foo: Foo = fixed_width::from_str("Foo").unwrap();
```
still get `Foo { bar: Bar }`

**Comment by twking7** [comment]
Hey, just got some time to take a look at this, and it's actually a bug. It should be fixed in the latest version, 0.4.1.

Let me know if that fixes it for you, and thanks for your help debugging!

Edit: Just noticed your PR - thanks for that! I didn't see it before I pushed the fix, but I appreciate you contributing.

**Comment by ongchi** [comment]
It's totally works! thank you 👍
